apps/api/src/domain/common/events/coscrad-event-handler.decorator.ts

```diff
@@ -0,0 +1,22 @@
+import { Maybe } from '../../../lib/types/maybe';
+import { NotFound } from '../../../lib/types/not-found';
+
+const COSCRAD_EVENT_HANDLER_META = '__COSCRAD_EVENT_HANDLER__';
+
+type EventHandlerScope = 'CREATE' | 'UPDATE';
+
+export type CoscradEventHandlerMeta = {
+    eventType: string;
+    scope: EventHandlerScope;
+    // Do we need to know the target aggregate type?
```

Comment:
No. Whatever is on the event (payload) will be matched against the instance when applying events.